main.go

```diff
@@ -143,18 +173,93 @@ func getLocator(ctx context.Context) (*ipfsgeoip.IPLocator, error) {
 
 	go lite.Bootstrap(ipfslite.DefaultBootstrapPeers())
 
-	// An exchange session will speed things up
 	return ipfsgeoip.NewIPLocator(lite.Session(ctx)), nil
 }
 
-func getMinerInfos(db *pg.DB) ([]minerInfo, error) {
+func getRouter(ctx context.Context) (*filecoinPeer, error) {
+	fp, err := setupFilecoinPeer(
+		ctx,
+		"mainnet",
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return fp, nil
+}
+
+func getActiveMiners(db *pg.DB, fp *filecoinPeer) (<-chan minerInfo, error) {
+	go fp.bootstrap()
+
+	// get active miners in last two weeks, i.e.
+	// miners making power claims
+	// access concurrently
+	var activeMiners []minerInfo
+	_, err := db.Query(&activeMiners, activeMinersQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	found := make(chan minerInfo)
+	var wg sync.WaitGroup
+	minersFound := 0
+
+	for i, activeMiner := range activeMiners {
+		i, activeMiner := i, activeMiner
+		wg.Add(1)
+
+		go func() {
```

Comment:
I am guessing that if we have 2000 active miners this is going to launch 2000 goroutines which are going to launch 2000 parallel dht queries.

Might be better to send that list of peer IDs to a channel and launch 50 workers to read from it and process each peer ID, so that we don't have more than 50 queries on flight.